iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift

```diff
@@ -292,117 +292,115 @@ class SwipeableInputAccessoryView: View, AttachmentViewControllerDelegate, UIGes
 
     func attachmentView(_ controller: AttachmentViewController, didSelect attachment: Attachment) {}
 
+    // MARK: - Pan Gesture Handling
+
+    private var previewView: PreviewMessageView?
+    private var initialPreviewOrigin: CGPoint?
+
+    func handle(pan: UIPanGestureRecognizer) {
+        guard self.shouldHandlePan() else { return }
+
+        // How far the preview view can be dragged left or right.
+        let maxXOffset: CGFloat = 100
+        // How far the preview view can be dragged vertically
+        let maxYOffset: CGFloat = SwipeableInputAccessoryView.maxHeight.half
+        let panOffset = pan.translation(in: nil)
+        // The percentage of the max y offset that the preview view has been dragged.
+        let progress = clamp(-panOffset.y/maxYOffset, 0, 1)
+
+        switch pan.state {
+        case .possible:
+            break
+        case .began:
+            self.handlePanBegan()
+        case .changed:
+            self.handlePanChanged(withOffset: panOffset, maxXOffset: maxXOffset, maxYOffset: maxYOffset)
+        case .ended:
+            self.handlePanEnded(withProgress: progress)
+        case .cancelled, .failed:
+            self.handlePanFailed()
```

Comment:
This function was getting too big so I split each case handler into its own function.